### PR TITLE
removed resource limits from  flannel template

### PR DESCRIPTION
**Pull request by vardhaman22**
related issue: https://github.com/rancher/rancher/issues/40178

### CODE REVIEW

**Review by kinarashah — CHANGES_REQUESTED** [review]
Ok to update existing template since `flannel_v0.19.2` is not yet released. 
But currently flannel v0.19.2 is used only for `>= v1.25` and t he original issue is to remove resource limits for Jan patches. 

So you would need to add and update flannel v0.14.0-rancher2 as well. Similar to https://github.com/rancher/kontainer-driver-metadata/pull/1067/files. Also, `v0.14.0-rancher2` template would be the same in both 2.6 and 2.7 to handle upgrades.

**Comment by vardhaman22** [comment]
> Ok to update existing template since `flannel_v0.19.2` is not yet released. But currently flannel v0.19.2 is used only for `>= v1.25` and t he original issue is to remove resource limits for Jan patches.
> 
> So you would need to add and update flannel v0.14.0-rancher2 as well. Similar to https://github.com/rancher/kontainer-driver-metadata/pull/1067/files. Also, `v0.14.0-rancher2` template would be the same in both 2.6 and 2.7 to handle upgrades.

updated the PR with suggested changes.

File: rke/templates/templates.go [hunk 1]
```diff
@@ -268,11 +272,12 @@ func getTemplates() map[string]string {
...
 
-		flannelv115:  FlannelTemplateV115,
-		flannelv116:  FlannelTemplateV116,
-		flannelv0140: FlannelTemplateV0_14_0,
-		flannelv0192: FlannelTemplateV0_19_2,
-		flannelv18:   FlannelTemplate,
+		flannelv115:          FlannelTemplateV115,
+		flannelv116:          FlannelTemplateV116,
+		flannelv0140:         FlannelTemplateV0_14_0,
+		flannelv0140Rancher2: flannelv0140Rancher2,
```
Comment (snasovich) [inline rke/templates/templates.go]: I'm pretty sure the value should be `FlannelTemplateV0_14_0Rancher2` here.

File: rke/templates/flannel_v0.14.0Rancher2.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
+package templates
```
Comment (snasovich) [inline rke/templates/flannel_v0.14.0Rancher2.go]: Please rename the file to `flannel_v0.14.0-rancher2.go` to follow the established naming convention, see https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.7/rke/templates

File: rke/templates/flannel_v0.14.0Rancher2.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+/*
+FlannelTemplateV0_14_0Rancher2 is based on upstream flannel v0.14.0
+Source: https://raw.githubusercontent.com/flannel-io/flannel/v0.14.0/Documentation/kube-flannel.yml
+
+Rancher Changelog:
+- Remove resource limits for kube-flannel DaemonSet
```
Comment (snasovich) [inline rke/templates/flannel_v0.14.0Rancher2.go]: Let's reference upstream PR here so it doesn't look like Rancher-specific change - https://github.com/flannel-io/flannel/pull/1694

**Review by snasovich — DISMISSED** [review]

**Comment by snasovich** [comment]
@vardhaman22 , dismissing my review to not block merging as I will be out till next week. As soon as @kinarashah approves following the fixes we're good to merge. Thank you!

File: rke/templates/templates.go (hunk 1 above)
Comment (vardhaman22) [reply rke/templates/templates.go]: updated

File: rke/templates/flannel_v0.14.0Rancher2.go (hunk 2 above)
Comment (vardhaman22) [reply rke/templates/flannel_v0.14.0Rancher2.go]: updated

File: rke/templates/flannel_v0.14.0Rancher2.go (hunk 3 above)
Comment (vardhaman22) [reply rke/templates/flannel_v0.14.0Rancher2.go]: updated

**Review by kinarashah — APPROVED** [review]

**Pull request merged**